btree/btree.c

```diff
@@ -2355,6 +2355,40 @@ static void bnode_op_fini(struct node_op *op)
  *  --------------------------------------------
  */
 
+/**
+ * @brief Proposed design for using btree node to hold embedded records with
+ * indirect addressing.
+ *
+ * Fixed format node = {
+ * 	header
+ * 	dir   = {
+ * 			ok1, ok2, ok3,..., okn,
+ * 			ov1, ov2, ov3,..., ovn
+ * 		}
+ * 	rec   = {
+ * 			(k1,v1), (k2,v2), (k3,v3),..., (kn,vn)
+ * 		}
+ * }
+ *
+ * nr_records = (node size - header size) /
+ * 			 (key size + val size + kptr size + vptr size)
+ *
+ * dir size   = nr_records * (kptr size + vptr size)
+ *
```

Comment:
Please describe how this node format is used. In case of inserts how does this work and in case of deletes how is this modified. Also how is the space of the deleted record reclaimed for new inserts. Please describe these.

Reply:
Done.